shipment_management/shipment_management/doctype/dti_shipment_note/dti_shipment_note.py

```diff
@@ -57,7 +57,15 @@ def on_submit(self):
 		frappe.db.set(self, 'fedex_status', ShipmentNoteOperationalStatus.InProgress)
 		frappe.db.set(self, 'tracking_number', self.box_list[0].tracking_number)
 
-	# def on_cancel(self):
+	def on_cancel(self):
+		tracking_ids_list = frappe.get_value("Sales Order", self.sales_order, "tracking_ids").split(',')
```

Comment:
```suggestion
		tracking_ids_list = frappe.db.get_value("Sales Order", self.sales_order, "tracking_ids").split(',')
```

Also you might have to strip every tracking id for better matching